packages/gatsby-plugin-sitemap/src/gatsby-node.js

```diff
@@ -26,7 +31,35 @@ exports.onPostBuild = async ({ graphql, pathPrefix }, pluginOptions) => {
     excludeOptions,
     pathPrefix
   )
-  serialize(queryRecords).forEach(u => map.add(u))
+  const urls = serialize(queryRecords)
 
-  return await writeFile(saved, map.toString())
+  if (!rest.sitemapSize || urls.length <= rest.sitemapSize) {
+    const map = sitemap.createSitemap(rest)
+    urls.forEach(u => map.add(u))
+    return writeFile(saved, map.toString())
+  }
+
+  const {
+    site: {
+      siteMetadata: { siteUrl },
+    },
+  } = queryRecords
+  return new Promise(resolve => {
+    // sitemapv-index.xml is default file name. (https://git.io/fhNgG)
```

Comment:
```suggestion
    // sitemap-index.xml is default file name. (https://git.io/fhNgG)
```

`sitemapv-index` -> `sitemap-index`?

Reply:
@muescha it's already merged, could you add a new pr? :)

Reply:
ok done -> #12512

Reply:
@muescha @wardpeet Thanks for that 😓